mycroft/skills/skill_manager.py

```diff
@@ -308,13 +310,18 @@ def _load_skill(self, skill_directory):
         return skill_loader if load_status else None
 
     def _get_skill_directories(self):
-        skill_glob = glob(os.path.join(self.msm.skills_dir, '*/'))
+        skill_glob = []
+        for directory in BaseDirectory.load_data_paths('mycroft/skills'):
```

Comment:
as mentioned in comment above we should support the old mechanism also, ideally core would check and monitor all skill directories instead of a single path, old location (configurable in .conf, can be set by user) + xdg

Reply:
What would you set the default location too then? My goal is to get rid of `/opt` by default, using just XDG accomplishes that.

What about `if old location exists in conf, use that (useful for dev purposes), otherwise use xdg`?

Reply:
we should support several directories by design, 
xdg allows :
-system level skills (all users) 
-skills for specific users

and .conf override allows:
- installation specific skills
- multiple non conflicting mycroft-core installs

Reply:
also note that this was originally planned but not implemented, the .conf contains

```
// Directory to look for user skills
"directory": "~/.mycroft/skills",
```

since you are touching this maybe add support for that (i still think data_dir is cleaner) or remove it from .conf since it's wrong info

Reply:
So I guess set default value to `{XDG}` which code checks and parses to proper XDG directories, and if something else just use that straight up?

Reply:
I'm thinking, this kinda conflicts with what I'm doing in #2578 as there I'm getting rid of the `data_dir` variable where possible. However, I do see value in allowing to keep that variable and allowing to set it to `{XDG}` by default so you can keep a dev setup and daily use setup separated. In your dev setup just set the variable to some other location and bam, completely separate configs and data/skills.
That is assuming you can use a different config for a dev setup altogether, which I'm not sure how to handle.

Reply:
also for backwards compatibility data_dir should no be deprecated (at least without a warning), i use it on ovos_utils and ovos_skills_manager for example

you raise a valid point about config, i usually edit the default .conf in mycroft-core for the specific dev setup, would be good to allow the user to set the path for user_config also, so by default it would also use `{XDG}` but individual installs could override it, but thats probably a separate PR

Reply:
So currently I move any skills found in the location as specified in the config to the new XDG _user_ location. Is that alright with you?

Reply:
IMO, avoid moving skills and data on existing installs. Too much room for deployment failure. Set it up so future installs will be XDG-compliant by default, and if you're monitoring multiple skill folders anyway, begin installing *new* skills in the XDG-compliant locations.

Just my two cents.

Reply:
I don't think I agree. That way we can never really get rid of `data_dir` as there might always be people who haven't moved over yet. For configuration I understand, people need to access those directly in the filesystem sometimes to edit them. But for data like this, nobody is manually editing anything in the skills. Maybe skill developers, but they can just find their skill at the new location.

Reply:
As discussed on Mattermost, using the XDG env variables properly you can keep a dev setup (or other Mycroft instances) separated from your regular installation. This will need proper documentation of course, but we have no need anymore for a config value anymore. Resolving this conversation.